Automattic/jetpack
# Protect: Fix Status and Scan_History option deletion on deactivation

## Description
Ensures that all relevant Protect status and history options are deleted when the plugin is deactivated.

## Proposed changes:

* Calls the parent `Status` class `delete_option` method from each child class within the deactivation method
* Adds a `Scan_History::delete_option` call to the same deactivation method

### Other information:

## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1346

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Generate a new Jurassic Ninja instance
* Install the Beta Tester and the latest stable version of Protect and let the first Protect scan complete
* Upgrade and let the initial Scan complete
* Verify that the following options exist:
   * `jetpack_protect_status`
   * `jetpack_protect_status_time`
   * `jetpack_scan_status`
   * `jetpack_scan_status_timestamp`
   * `jetpack_scan_history`
   * `jetpack_scan_history_timestamp`
* Deactivate the plugin and verify that the are not removed
* Reactivate the plugin
* Upgrade Protect from the Beta Tester to this branch
* Deactivate the plugin and verify that the are deleted
